File: src/C#/CreationalPatterns/Singleton/VotingMachine/VotingMachineV4.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace VotingMachine
{
    public sealed class VotingMachineV4
    {
        private int totalVotes = 0;

        private static readonly Lazy<VotingMachineV4> votingMachineInstance = new Lazy<VotingMachineV4>(() => new VotingMachineV4());

        private VotingMachineV4()
        {

        }

        public int TotalVotes => totalVotes;

        public static VotingMachineV4 VotingMachineInstance => votingMachineInstance.Value;

        public void RegisterVote()
        {
            totalVotes++;
            Console.WriteLine($"Registered Vote #{totalVotes}");
        }
    }
}

File: src/Java/CreationalPatterns/Singleton/VotingMachine/src/VotingMachineV4.java
import java.util.concurrent.atomic.AtomicInteger;

public final class VotingMachineV4 {
    private final AtomicInteger totalVotes = new AtomicInteger(0);

    private static final VotingMachineV4 votingMachineInstance = new VotingMachineV4();

    private VotingMachineV4() {

    }

    public int getTotalVotes() {
        return totalVotes.get();
    }

    public static VotingMachineV4 getVotingMachineInstance() {
        return votingMachineInstance;
    }

    public void registerVote() {
        int currentVotes = totalVotes.incrementAndGet();
        System.out.println("Registered Vote #" + currentVotes);
    }
}
